### PR TITLE
kitty: 0.24.2 -> 0.24.4

### PR DESCRIPTION
**Pull request by Luflosi**
###### Motivation for this change
https://github.com/kovidgoyal/kitty/releases/tag/v0.24.4

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Luflosi** [comment]
A test fails on macOS. I'll need to investigate.

**Comment by jtrees** [comment]
FYI: Looks like 0.24.4 has been released already.

**Comment by page-down** [comment]
@Luflosi 

This should be updated to `fish2` (the second version of fish completion).
In terms of style, the `+complete` namespace is recommended. (Without the space in between. (Just a suggestion, consistent with all styles in the official documentation.)

```text
installShellCompletion --cmd kitty \
...
--fish <("$out/bin/kitty" + complete setup fish) \
```

**Comment by jtrees** [comment]
I'm not a nix pro and not really familiar with the nixpkgs guidelines but this PR looks fine to my eyes. I'm also able to compile kitty and it runs. Thanks!

**Comment by Luflosi** [comment]
Yep, seems to work now. I'll wait for one more response from upstream and modify the commit message to credit @page-down, then I'll mark this as ready to review.
Thanks @page-down for your suggestion.

File: pkgs/applications/terminal-emulators/kitty/default.nix [hunk 1]
```diff
@@ -18,19 +18,23 @@
   libpng,
   python3,
   zlib,
+  bashInteractive,
+  zsh,
+  fish,
+  fetchpatch,
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/kitty/default.nix:24]:
```suggestion
, bashInteractive
, zsh
, fish
, fetchpatch
```

File: pkgs/applications/terminal-emulators/kitty/test.patch [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
++            pty.wait_till(lambda: pty.screen_contents().count(rps1) == 2)
++            print(pty.screen_contents())
++            self.ae(pty.last_cmd_output(), q)
+             self.ae(pty.callbacks.titlebuf[-1], '~')
+             pty.callbacks.clear()
+             pty.send_cmd_to_child('mkdir test && ls -a')
+diff --git a/shell.nix b/shell.nix
+index 4e5744fd..78c83dd8 100644
+--- a/shell.nix
++++ b/shell.nix
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/kitty/test.patch:20]: Can we use the shell at all?

Comment (Luflosi) [reply pkgs/applications/terminal-emulators/kitty/test.patch:20]: No, I was just testing. I want to see the the contents of a variable, see https://github.com/kovidgoyal/kitty/issues/4782. I'll remove the commit and finish this PR once I have my answer.

**Comment by Luflosi** [comment]
@leungbk 21.05? You probably mean 22.05, right?

**Comment by leungbk** [comment]
@Luflosi Whoa sorry, I was browsing this PR yesterday without intending to modify it and must have unwittingly done something. I cannot seem to remove the Milestone tag for some reason, possibly bc I am on phone.

**Comment by Luflosi** [comment]
No problem, I removed it from the milestone.

**Pull request merged**